bin/warnhist

```diff
@@ -67,29 +86,196 @@ sub do_help {
     print "\t\t<n> can be a single value, a range, or a comma separated list\n";
     print "\t\tFor example: '0' or '0,4' or '8-10' or '0,2-4,8-10,13'\n";
     print "\t-F\tDisplay warnings for all files\n";
-    print "\t-s <warning string list>\tDisplay files for warnings which contain a string, <warning string list>\n";
+    print "\t-s <warning string list>  Display files for warnings which contain a\n";
+    print "\t\tstring, <warning string list>\n";
     print "\t\t<warning string list> is a comma separated list, with no spaces\n";
-    print "\t\tFor example: 'Wunused-dummy-argument' or 'Wunused-dummy-argument,Wunused-variable'\n";
-    print "\t-S <file string list>\tDisplay warnings for files which contain a string, <file string list>\n";
+    print "\t\tFor example: 'Wunused-dummy-argument' or\n";
+    print "\t\t'Wunused-dummy-argument,Wunused-variable'\n";
+    print "\t-S <file string list>  Display warnings for files which contain a\n";
+    print "\t\tstring, <file string list>\n";
     print "\t\t<file string list> is a comma separated list, with no spaces\n";
     print "\t\tFor example: 'H5Fint' or 'H5Fint,H5Gnode'\n";
     print "\t-l\tDisplay line numbers for file/warning\n";
     print "\t-u\tLeave 'unique' types in warnings, instead of genericizing them\n";
     print "\t-i <name list>\tIgnore named files, <name list>\n";
     print "\t\t<name list> is a comma separated list, with no spaces\n";
     print "\t\tFor example: 'H5LTparse' or 'H5LTparse,H5LTanalyze'\n";
+    print "\t-d\tCompute warning density for compiled source files.  Paths to the\n";
+    print "\t\troot of a directory containing source may be provided with the\n";
+    print "\t\t'-p <path list>' option.  If the path list is given, only those\n";
+    print "\t\tdirectories are scanned for source files.  If the path list\n";
+    print "\t\toption is not given, the current working directory is scanned.\n";
+    print "\t-p <path list>\tPaths to search for compiled files.  Compiled files\n";
+    print "\t\tare only used when computing warning density and are not\n";
+    print "\t\tnecessary for just analyzing warnings in build output.\n";
+    print "\t\t<path list> is a comma separated list, with no spaces\n";
+    print "\t\tFor example: '/home/koziol/hdf5' or '.,~/dev/hdf5,~/dev/build'\n";
+    print "\t-q\tSuppress warning output\n";
```

Comment:
I also added a '-q' ("quiet") option to suppress warnings from the script itself.